master/buildbot/config.py

```diff
@@ -541,6 +541,10 @@ def load_slaves(self, filename, config_dict):
             if sl.slavename in ("debug", "change", "status"):
                 msg = "slave name '%s' is reserved" % sl.slavename
                 error(msg)
+            
+            if not util.identifiers.isIdentifier(50, sl.slavename):
+                msg = "Buildslave name %r is not a 50-character identifier" % (sl.slavename,)
+                error(msg)
```

Comment:
Actually I copied error messages from https://github.com/buildbot/buildbot/blob/master/master/buildbot/data/buildslaves.py#L115 . I can change both of them, but I see that isIdentifier function does not control only length of slavename. Slavename can also break some other rules. May it be good to print all of those rules, if name breaks one of them?